src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.AdjustmentRule.cs

```diff
@@ -129,8 +128,7 @@ public sealed class AdjustmentRule : IEquatable<AdjustmentRule?>, ISerializable,
             internal bool IsEndDateMarkerForEndOfYear() =>
                 !NoDaylightTransitions &&
                 DaylightTransitionEnd.Month == 1 && DaylightTransitionEnd.Day == 1 &&
-                DaylightTransitionEnd.TimeOfDay.TimeOfDay.Ticks < TimeSpan.TicksPerSecond && // < 12:00:01 AM
-                _dateStart.Year == _dateEnd.Year;
```

Comment:
Do you know why this extra check was there in the first place?

Reply:
It was done long ago to minimize the risk on the Full Framework and we didn't have any case breaking this. But now we got the case that broke it.